src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,478 @@
+(ns basilisp.contrib.nrepl-server
+  "A port of nbb's nREPL server implementation to basilisp.
+
+  https://github.com/babashka/nbb/blob/bca8b5017a06768eb35d02a2d6233ca9c6c2f692/src/nbb/impl/nrepl_server.cljs"
+  (:require [basilisp.contrib.bencode :as bc]
+            [basilisp.string :as str])
+  (:import [basilisp.__version__ :as ver]
+           [basilisp.lang.runtime :as rt]
+           socketserver
+           sys
+           traceback
+           uuid))
+
+(def ^:private log-levels
+  "A compile time set used to determine which log levels should be
+  included in the output."
+  #{:warn})
+
+;; helper compile time macros to log ``values`` according to the level
+;; as filtered by `log-levels`.
+(defmacro ^:private debug [& values]
+  (when (some #{:debug} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "debug") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private info [& values]
+  (when (some #{:info} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "info") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private warn [& values]
+  (when (some #{:warn} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "warn") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private error [& values]
+  ;; we want error to always output to `sys/stderr` unfiltered.
+  `(.write sys/stderr (str ~(keyword (str *ns*) "error") " " (str/join " " [~@values]) \newline)))
```

Comment:
Basilisp does ship with a [Python logger](https://github.com/basilisp-lang/basilisp/blob/main/src/basilisp/logconfig.py) which I would love to continue using if possible so that it's all configurable using a unified group of settings.

I don't want to let perfect be the enemy of good though, so we could also do this in a follow-up PR if this work ends up being more substantial than I am estimating.

Reply:
Right, I've changed to use the python logger. In this early stage, I thought it was a good idea to always display errors on the server process, and thus I've left the `error` fn as is, but can also change it to the logger if are feeling otherwise.

Reply:
This seems ok for now. I think we could ultimately enable just this logger for error level regardless of the settings of other loggers, but Python logging configuration can be fairly complex.

Reply:
ok thanks, I left it as is for the moment being.